lib/openstack_query/queries/query_wrapper.py

```diff
@@ -0,0 +1,130 @@
+    ):
+        self._query_results = []
+
+        self.runner = None
+        self.output = QueryOutput(prop_handler)
+        self.builder = QueryBuilder(
+            prop_handler,
+            client_side_handlers,
+            server_side_handler,
+        )
+
+    def select(self, *props: Enum):
+        """
+        Public method used to 'select' properties that the query will return the value of.
+        Mutually exclusive to returning objects using select_all()
+        :param props: one or more properties to collect described as enum
+        """
+
+        # is an idempotent function
+        # an be called multiple times with should aggregate properties to select
+        if not props:
+            raise ParseQueryError("provide at least one property to select")
+
+        self.output.parse_select(*props, select_all=False)
+        return self
+
+    def select_all(self):
+        """
+        Public method used to 'select' all properties that are available to be returned
+        Mutually exclusive to returning objects using select_all()
+
+        Overrides all currently selected properties
+        returns list of properties currently selected
+        """
+        self.output.parse_select(select_all=True)
+        return self
+
+    def where(self, preset: Enum, prop: Enum, **kwargs: Optional[Dict[str, Any]]):
+        """
+        Public method used to set the conditions for the query.
+        :param preset: Name of query preset to use
+        :param prop: Property that the query preset will be used on
+        :param kwargs: a set of optional arguments to pass along with the preset - property pair
+            - these kwargs are dependent on the preset given
+        """
+        self.builder.parse_where(preset, prop, kwargs)
+        return self
+
+    def sort_by(self, sort_by: Enum, reverse=False):
+        """
+        Public method used to configure sorting results
+        :param sort_by: name of property to sort by
+        :param reverse: False is sort by ascending, True is sort by descending, default False
+        """
+        raise NotImplementedError
+
+    def group_by(self, group_by: Enum):
+        """
+        Public method used to configure grouping results.
+        :param group_by: name of the property to group by
+        """
+        raise NotImplementedError
+
+    def run(
+        self, cloud_account: str, from_subset: Optional[List[Any]] = None, **kwargs
+    ):
+        """
+        Public method that runs the query provided and outputs
+        :param cloud_account: The account from the clouds configuration to use
+        :param from_subset: A subset of openstack resources to run query on instead of querying openstacksdk
+        :param kwargs: keyword args that can be used to configure details of how query is run
+            - valid kwargs specific to resource
+        """
+        filter_func = self.builder.filter_func()
+        filter_kwargs = self.builder.server_side_filters()
```

Comment:
Nit, these names need updating to something like `local_filters` and `server_filters` following on from the class renames